byterun/debugger.c

```diff
@@ -121,7 +121,7 @@ static void open_connection(void)
 #endif
   if (dbg_socket == -1 ||
       connect(dbg_socket, &sock_addr.s_gen, sock_addr_len) == -1){
-    caml_fatal_error_arg2 ("cannot connect to debugger at %s\n", dbg_addr,
+    caml_fatal_error_arg2 ("cannot connect to debugger at %s\n", (dbg_addr ? dbg_addr : "(none)"),
```

Comment:
(related to the change above)

Reply:
Because `dbg_addr` is now heap-allocated, we need to know when to free it:

https://github.com/ocaml/ocaml/pull/1200/files#diff-b5ec566db11da31290ab22a114e24fe4R180

Of course, if `caml_debugger_init` is only called once, then we don't care about this.  `caml_debugger_init` is called from `caml_main` and other similar functions.  Can they be called more than once in the lifetime of a program ?

Reply:
Ah, I understand - this makes sense as it is. It shouldn't be called more than once without tearing the runtime down in between, I think.